File: MgDev/UnitTest/WebTier/DotNet/TestMapGuideApi/DrawingService/DrawingServiceOperationExecutor.cs
using OSGeo.MapGuide.Test.Common;
using SqliteDotNet;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OSGeo.MapGuide.Test.Operations
{
    /// <summary>
    /// The base class of all test executors using the MgDrawingService
    /// </summary>
    /// <typeparam name="T">The test executor subclass type</typeparam>
    public abstract class DrawingServiceOperationExecutor<T> : PlatformApiTestExecutor
    {
        protected MgDrawingService _drawingService;

        protected DrawingServiceOperationExecutor(MgDrawingService drawSvc, string unitTestVm)
            : base(typeof(T).Name.ToUpper(), ApiTypes.Platform, unitTestVm)
        {
            _drawingService = drawSvc;
        }
    }
}

File: MgDev/UnitTest/WebTier/Java/src/org/osgeo/mapguide/test/mapguide/operations/drawing/DrawingServiceOperationExecutor.java
package org.osgeo.mapguide.test.mapguide.operations.drawing;

import org.osgeo.mapguide.*;
import org.osgeo.mapguide.test.*;
import org.osgeo.mapguide.test.common.*;

import com.almworks.sqlite4java.*;

public abstract class DrawingServiceOperationExecutor extends PlatformApiTestExecutor
{
    protected MgDrawingService _drawingService;

    protected DrawingServiceOperationExecutor(String opName, MgDrawingService drawSvc, String unitTestVm)
    {
        super(opName, ApiTypes.Platform, unitTestVm);
        _drawingService = drawSvc;
    }
}